fluentbit-kafka-fluentd/README.md

```diff
@@ -0,0 +1,143 @@
+# Face to Face 2020/03 - Fluentbit -> Kafka -> Fluentd -> Elasticsearch
+
+## What?
+The purpose of this hackathon item was to investigate a new collector/aggregator solution for Cluster Logging.
+
+## Why?
+Currently Fluentd takes up decent amount of space on the host nodes in order to collect, enrich, and ship logs.
```

Comment:
s/space/RAM?

Reply:
i'll expand on that, i means both CPU and RAM

Reply:
Would it make sense to run a quick benchmark and get the two fluents to process the same amount of logs, just dump them into stdout and compare CPU utilization and RAM footprints?

Reply:
Here's a quick 'n dirty comparison of CPU and RAM footprints of Fluent Bit and Fluentd, each tailing a JSON log file and dumping into stdout and ultimately /dev/null. The log generator is a slightly hacked https://github.com/ViaQ/logging-load-driver/blob/master/loader so it produces JSON.

Fluent Bit:
```
$ ./loader | pv -r > loader.out
[1.40MiB/s]
$ ~/f2f/fluent-bit-1.3.10/build/bin/fluent-bit -i tail -p path=/home/syedriko/go/src/github.com/openshift/ViaQ/logging-load-driver/loader.out -o stdout > /dev/null
$ top -p 18473,18475    
PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND 
18473 syedriko  20   0  224.3m  13.6m   7.4m R  99.3   0.1   1:15.50 python            
18475 syedriko  20   0   25.8m   8.2m   5.0m S   1.0   0.1   0:00.66 fluent-bit            
```

Fluentd:
```
$ ./loader | pv -r > loader.out
[1.37MiB/s]

$ ~/bin/fluentd -c ./fluent.conf -v > /dev/null
$ cat fluent.conf
<source>
  @type tail
  path /home/syedriko/go/src/github.com/openshift/ViaQ/logging-load-driver/loader.out
  pos_file /home/syedriko/go/src/github.com/openshift/ViaQ/logging-load-driver/loader.out.pos
  tag viaq_loader
  <parse>
    @type json
  </parse>
</source>

<match *>
  @type stdout
</match>
$

$ top -p 18296,18293
    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                        
  18296 syedriko  20   0  224.3m  13.5m   7.2m R 100.0   0.1   1:33.91 python                                        
  18293 syedriko  20   0  414.3m  57.3m   8.8m S   5.7   0.4   0:05.89 ruby-mri
```

Reply:
@syedriko was that python process the log generator?

Reply:
Yes it was. Its high CPU utilization was from it generating random log lines. It can be told to log deterministic messages with "--payload-gen fixed", with much lower CPU load and much higher throughput.